src/components/DualListBox/DualListBox.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from "react";
+import * as Styled from "./styled";
+import { CandidateRenderer } from "./internal/CandidateRenderer";
+import { SelectedList } from "./internal/SelectedList";
+import Divider from "../Divider/Divider";
+import { useTheme } from "../../themes";
+
+export type ItemBase = {
+  id: string;
+};
+
+export type DualListBoxItemSelectedWithToggle = ItemBase & { checked: boolean };
+export type DualListBoxItemSelectedWithoutToggle = ItemBase & {
+  checked?: undefined;
+};
```

Comment:
nits
こんなかんじかな...?
`WithoutToggle`はなるべく無くすような命名にした方が読み疲れしなさそうだね。

```suggestion
export type DualListBoxItemSelectedWithToggle = ItemBase & { checked: boolean };
export type DualListBoxItemSelected = ItemBase
```

Reply:
checked は明記しないとダメじゃないですか？
? のオプショナルと undefined の明記は意味が違うので。

Reply:
あーたしかにそうだね！